sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/AbstractAggregator.java

```diff
@@ -38,12 +46,37 @@ public void mergeToAndReset(Aggregator other) {
   abstract void doMergeAndReset(Aggregator aggregator);
 
   @Override
-  public void recordLong(long value) {
-    throw new UnsupportedOperationException("This Aggregator does not support long values");
+  public final void recordLong(long value) {
+    hasRecordings.set(true);
```

Comment:
Isn't it better to set after recording since that's when it actually "has recordings"?

Reply:
I've gone back and forth several times on this, trying to figure out which leads to the least-bad race condition when it happens. I think you're right, though; I'll swap it around.

Reply:
done. thanks for the close look!